corehq/apps/users/tests/retire.py

```diff
@@ -0,0 +1,135 @@
+from django.test import TestCase
+import mock
+
+import uuid
+from xml.etree import ElementTree
+from corehq.apps.users.models import CommCareUser
+from corehq.apps.hqcase.utils import submit_case_blocks
+from casexml.apps.case.mock import CaseBlock
+from casexml.apps.case.models import CommCareCase
+from casexml.apps.case.tests.util import delete_all_cases, delete_all_xforms
+from couchforms.models import XFormInstance
+
+
+class RetireUserTestCase(TestCase):
+
+    def setUp(self):
+        self.domain = 'test'
+        self.username = "fake-person@test.commcarehq.org"
+        self.other_username = 'other-user@test.commcarehq.org'
+        self.password = "s3cr3t"
+
+        self.commcare_user = CommCareUser.create(self.domain, self.username, self.password)
+        self.commcare_user.save()
+
+        self.other_user = CommCareUser.create(self.domain, self.other_username, self.password)
+        self.other_user.save()
+
+    def tearDown(self):
+        for user in CommCareUser.all():
+            user.delete()
+        delete_all_cases()
+        delete_all_xforms()
+
+    @mock.patch("casexml.apps.case.cleanup.rebuild_case")
+    def test_rebuild_cases_with_new_owner(self, rebuild_case):
+        """
+            If cases have a different owner to the person who submitted it
+            rebuild it when the submitter is retired.
+        """
+
+        case_id = uuid.uuid4().hex
+        caseblock = CaseBlock(
+            create=True,
+            case_id=case_id,
+            owner_id=self.commcare_user._id,
+            user_id=self.commcare_user._id,
+        )
+        casexml = ElementTree.tostring(caseblock.as_xml())
+        form_id = submit_case_blocks(casexml, self.domain)
+        form = XFormInstance.get(form_id)
+
+        form.form['meta']['userID'] = self.other_user._id
```

Comment:
This is cheating (editing the form directly). You should rather use `self.other_user._id` as the value for `user_id` when you create the caseblock (line 46)